pkg/util/pki/generate.go

```diff
@@ -1,18 +1,110 @@
 package pki
 
 import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
+	"fmt"
+
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+)
+
+var ellipticCurves = map[int]elliptic.Curve{
+	224: elliptic.P224(),
+	256: elliptic.P256(),
+	384: elliptic.P384(),
+	521: elliptic.P521(),
```

Comment:
Is it okay to reuse these (from a security perspective)
Also are they thread-safe? We run 5 Certificate workers, so there's a these could be used in parallel.

Reply:
> Is it okay to reuse these (from a security perspective)

From the go source https://golang.org/src/crypto/elliptic/elliptic.go?s=11343:11360#L367, this is calculated once and cached. Just to ensure I didn't miss anything, I've changed it from using a map to calculating the curve when needed.

> Also are they thread-safe? We run 5 Certificate workers, so there's a these could be used in parallel.

Afaik, in go, its ok to have concurrent reads on a map with no writer. I've removed the map implementation so this shouldn't be a problem anymore